controllers/teleport_controller/teleport_controller.cpp

```diff
@@ -0,0 +1,138 @@
+// File:          teleport_controller.cpp
+// Date:          23/03/2021
+// Description:   Method to teleport robot around the playing field for the purposes of
+//                    gathering training data
+// Author:        Lachlan Court
+// Modifications:
+
+
+#include <RobotisOp2GaitManager.hpp>
+#include <RobotisOp2MotionManager.hpp>
+#include <webots/Supervisor.hpp>
+#include <iostream>
+#include <math.h>
+#include <time.h> 
+
+// All the webots classes are defined in the "webots" namespace
+using namespace webots;
+
+/*
+// Rotations matrix for 90 degree rotations
+static const int rotationsSize = 4;
+static const double rotations[rotationsSize][4] = {{1, 0, 0, 1.57081},
+                                                  {0.577353, 0.577345, 0.577353, 2.0944},
+                                                  {1.69526e-09, 0.707103, 0.707111, 3.14159},
+                                                  {-0.577352, 0.577347, 0.577352, -2.09441}};
+
+*/
+
+// Rotations matrix for 45 degree rotations
+static const int rotationsSize = 8;                                     
```

Comment:
numeric constants would normally use screaming snake case, but we probably won't keep this one (`ROTATIONS_SIZE` would be preferred if we did keep it). You do this so that the developer knows when they attempt to write to a variable in that case that they are doing something wrong.

Reply:
What do you mean by we won't keep it? But yes I did know that and forgot to do it 🤦

Reply:
we probably won't keep it because its purpose is for a loop bound, and you should use range-based for loops for the standard containers (std::array, std::vector, etc.), or their `.size()` methods if you need the size for some reason. [.size() for std::vector on cppreference](https://en.cppreference.com/w/cpp/container/vector/size) (an incredible website btw)